### PR TITLE
Add enabled for FileWatchManager and FileWatchLogic.  Add call to HandleProjectLoaded inside SaveProject

### PR DESCRIPTION
**Pull request by kaltinril**
Attempt at #405

### CODE REVIEW

**Comment by kaltinril** [comment]
@vchelaru I couldn't get it to work correctly.  

1. Couldn't figure out how to force the File Paths watching to get emptied out on New Project
2. I was able to get a Save Project to call the functionality that gets the root project path
3. However... when I added a screen and components, it didn't have standard or screen or component folder it was watching, so I didn't do something correctly in this PR.

So there must be other locations.

**Comment by vchelaru** [comment]
I think the changes you made were good. One of the biggest problems with the code (as it was before the PR) is that it was not using the plugin pattern that new code is using, so like you said the code was spread across lots of files and the flow was hard to follow.

I moved most of the code reacting to project load/unloads/renames into the plugin, and also moved some files related to file watch into the plugin folder as well. 

I'll be happy to answer any questions or help resolve confusion.
